openmetadata-service/src/main/java/org/openmetadata/service/search/elasticsearch/ElasticSearchClient.java

```diff
@@ -1953,6 +1955,42 @@ public void createEntity(String indexName, String docId, String doc) {
     }
   }
 
+  @Override
+  public void createEntities(String indexName, List<Map<String, String>> docsAndIds)
+      throws IOException {
+    if (isClientAvailable) {
+      BulkRequest bulkRequest = new BulkRequest();
+      for (Map<String, String> docAndId : docsAndIds) {
+        Map.Entry<String, String> entry = docAndId.entrySet().iterator().next();
+        IndexRequest indexRequest =
+            new IndexRequest(indexName)
+                .id(entry.getKey())
+                .source(entry.getValue(), XContentType.JSON);
+        bulkRequest.add(indexRequest);
+      }
+      bulkRequest.setRefreshPolicy(WriteRequest.RefreshPolicy.IMMEDIATE);
+      ActionListener<BulkResponse> listener =
+          new ActionListener<BulkResponse>() {
+            @Override
+            public void onResponse(BulkResponse bulkItemResponses) {
+              if (bulkItemResponses.hasFailures()) {
+                LOG.error(
+                    "Failed to create entities in ElasticSearch: {}",
+                    bulkItemResponses.buildFailureMessage());
+              } else {
+                LOG.debug("Successfully created {} entities in ElasticSearch", docsAndIds.size());
+              }
+            }
+
+            @Override
+            public void onFailure(Exception e) {
+              LOG.error("Failed to create entities in ElasticSearch", e);
```

Comment:
we may want to throw the exception back here to let the transaction know to revert the DB if there is unrecoverable exception

Reply:
@harshach I try to implement and did a bit of investigation as the ops don't get rolled back (more details [here](https://github.com/open-metadata/OpenMetadata/issues/19369)). Let me know if you have any thoughts about it.